### PR TITLE
Minimal install

### PR DESCRIPTION
**Pull request by liamfallon**

### CODE REVIEW

**Review by lapentad — APPROVED** [review]
/approve

File: developer-guide/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Nephio Developer Guide
+======================
+
+The developer guide is for anyone wanting to either write code for the Nephio project. THis document is written as an extension of the  [Kubernetes Developer Guides](https://github.com/kubernetes/community/tree/master/contributors/devel#the-process-of-developing-and-contributing-code-to-the-kubernetes-project) and therefore covers topics that are specific to Nephio development.
```
Comment (electrocucaracha) [inline developer-guide/README.md:4]:
```suggestion
The developer guide is for anyone wanting to either write code for the Nephio project. This document is written as an extension of the  [Kubernetes Developer Guides](https://github.com/kubernetes/community/tree/master/contributors/devel#the-process-of-developing-and-contributing-code-to-the-kubernetes-project) and therefore covers topics that are specific to Nephio development.
```

File: developer-guide/README.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+* How to set up a [Minimal Development Environment]minimal-environment.md) defines common terminology used in the Nephio project.
```
Comment (electrocucaracha) [inline developer-guide/README.md:6]:
```suggestion
* How to set up a [Minimal Development Environment](minimal-environment.md) defines common terminology used in the Nephio project.
```

File: developer-guide/minimal-environment [hunk 3]
```diff
@@ -0,0 +1,139 @@
+
+Minimal Environment install for development
+===========================================
+
+The following environment install works on a Macbook Pro M1 to get the nephio-operators running in vscode talking to a local kind   cluster running in Docker. Note that depending on what aprt of Nephio you are working on, you may wish to install less or more of the components below. It should work on other environments with appropriate tweaking.
```
Comment (electrocucaracha) [inline developer-guide/minimal-environment:5]:
```suggestion
The following environment install works on a Macbook Pro M1 to get the nephio-operators running in vscode talking to a local kind cluster running in Docker. Note that depending on what part of Nephio you are working on, you may wish to install less or more of the components below. It should work on other environments with appropriate tweaking.
```

File: developer-guide/minimal-environment [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+
+kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/nephio-controllers nephio-controllers
+find nephio-controllers/crd/bases/*.yaml -exec kubectl apply -f {} \;
+
+popd >/dev/null || exit
+```
+
+## Installation steps:
+
+1. Install Docker using the appropriate method for you system.
```
Comment (electrocucaracha) [inline developer-guide/minimal-environment:39]:
```suggestion
1. [Install Docker](https://docs.docker.com/engine/install/) using the appropriate method for you system.
```

File: developer-guide/minimal-environment [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+2. Install kind using the appropriate method for you system (homebrew on mac)
```
Comment (electrocucaracha) [inline developer-guide/minimal-environment:41]:
```suggestion
2. [Install kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) using the appropriate method for you system (homebrew on mac)
```

File: developer-guide/minimal-environment [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+```sh
+brew install kind
+```
```
Comment (electrocucaracha) [inline developer-guide/minimal-environment:45]: Providing a pointer to the kind installation, the following instruction can be omitted 
```suggestion
```

File: developer-guide/minimal-environment [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+9. Load the Nephio CRDs
+
+```sh
+kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/nephio-controllers nephio-controllers
+ls nephio-controllers/crd/bases/*.yaml | xargs -n1 kubectl apply -f
+```
+
+# Connect to Gitea on your browser
+
+COnnecting to Gitea allows you to see the actions that Nephio takes on Gitea.
```
Comment (electrocucaracha) [inline developer-guide/minimal-environment:100]:
```suggestion
Connecting to Gitea allows you to see the actions that Nephio takes on Gitea.
```

File: developer-guide/README.md (hunk 1 above)
Comment (liamfallon) [reply developer-guide/README.md]: Done

File: developer-guide/README.md (hunk 2 above)
Comment (liamfallon) [reply developer-guide/README.md]: Done

File: developer-guide/minimal-environment (hunk 4 above)
Comment (liamfallon) [reply developer-guide/minimal-environment]: done

File: developer-guide/minimal-environment (hunk 5 above)
Comment (liamfallon) [reply developer-guide/minimal-environment]: done

File: developer-guide/minimal-environment (hunk 6 above)
Comment (liamfallon) [reply developer-guide/minimal-environment]: Done

File: developer-guide/minimal-environment (hunk 7 above)
Comment (liamfallon) [reply developer-guide/minimal-environment]: Done

File: developer-guide/minimal-environment (hunk 3 above)
Comment (liamfallon) [reply developer-guide/minimal-environment]: done

**Comment by liamfallon** [comment]
@electrocucaracha thanks for the review and sorry for all the typos.

File: developer-guide/minimal-environment [hunk 8]
```diff
@@ -0,0 +1,135 @@
...
+
+popd >/dev/null || exit
+```
+
+## Installation steps:
+
+1. [Install Docker](https://docs.docker.com/engine/install/) using the appropriate method for you system.
+
+2. [Install kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) using the appropriate method for you system (homebrew on mac)
+
```
Comment (tomkivlin) [inline developer-guide/minimal-environment:42]: I think a step is needed here to install kpt - it's not included in the referenced Kubernetes [local development guide](https://github.com/kubernetes/community/blob/master/contributors/devel/development.md) so should be here, I think.

Comment (tomkivlin) [reply developer-guide/minimal-environment:42]: e.g. 

```
3. [Install kpt cli](https://kpt.dev/installation/kpt-cli) using the appropriate method for your system (e.g. homebrew on mac)
```

File: developer-guide/minimal-environment [hunk 9]
```diff
@@ -0,0 +1,135 @@
...
+find nephio-controllers/crd/bases/*.yaml -exec kubectl apply -f {} \;
+
+popd >/dev/null || exit
+```
+
+## Installation steps:
+
+1. [Install Docker](https://docs.docker.com/engine/install/) using the appropriate method for you system.
+
+2. [Install kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) using the appropriate method for you system (homebrew on mac)
```
Comment (tomkivlin) [inline developer-guide/minimal-environment:41]:
```suggestion
2. [Install kind](https://kind.sigs.k8s.io/docs/user/quick-start/#installation) using the appropriate method for your system (homebrew on mac)
```

File: developer-guide/minimal-environment [hunk 10]
```diff
@@ -0,0 +1,135 @@
...
+
+kpt pkg get https://github.com/nephio-project/nephio-example-packages/tree/main/nephio-controllers nephio-controllers
+find nephio-controllers/crd/bases/*.yaml -exec kubectl apply -f {} \;
+
+popd >/dev/null || exit
+```
+
+## Installation steps:
+
+1. [Install Docker](https://docs.docker.com/engine/install/) using the appropriate method for you system.
```
Comment (tomkivlin) [inline developer-guide/minimal-environment:39]:
```suggestion
1. [Install Docker](https://docs.docker.com/engine/install/) using the appropriate method for your system.
```

File: developer-guide/minimal-environment [hunk 11]
```diff
@@ -0,0 +1,135 @@
+
+Minimal Environment install for development
+===========================================
+
+The following environment install works on a Macbook Pro M1 to get the nephio-operators running in vscode talking to a local kind cluster running in Docker. Note that depending on what part of Nephio you are working on, you may wish to install less or more of the components below. It should work on other environments with appropriate tweaking.
```
Comment (tomkivlin) [inline developer-guide/minimal-environment:5]: I've confirmed this is working as documented using vscode on Windows connecting through SSH to Ubuntu 22.04 (where kind runs).

```suggestion
The following environment install works on a Macbook Pro M1 or via SSH to Ubuntu 22.04 to get the nephio-operators running in vscode talking to a local kind cluster running in Docker. Note that depending on what part of Nephio you are working on, you may wish to install less or more of the components below. It should work on other environments with appropriate tweaking.
```

File: developer-guide/minimal-environment (hunk 8 above)
Comment (liamfallon) [reply developer-guide/minimal-environment:42]: I added this step, thanks @tomkivlin  for the review.

Comment (tomkivlin) [reply developer-guide/minimal-environment]: Thanks @liamfallon

**Comment by johnbelamaric** [comment]
/approve
/lgtm

Nice!

**Comment by gvbalaji** [comment]
@liamfallon thanks a lot. This requires a rebase before merging.

File: developer-guide/README.md [hunk 12]
```diff
@@ -0,0 +1,6 @@
+Nephio Developer Guide
+======================
+
+The developer guide is for anyone wanting to either write code for the Nephio project. This document is written as an extension of the  [Kubernetes Developer Guides](https://github.com/kubernetes/community/tree/master/contributors/devel#the-process-of-developing-and-contributing-code-to-the-kubernetes-project) and therefore covers topics that are specific to Nephio development.
```
Comment (efiacor) [inline developer-guide/README.md:4]: Can this be omitted?

File: developer-guide/minimal-environment.md [hunk 13]
```diff
@@ -0,0 +1,137 @@
+
+Minimal Environment install for development
+===========================================
+
+The following environment install works on a Macbook Pro M1 or via SSH to Ubuntu 22.04 to get the nephio-operators running in vscode talking to a local kind cluster running in Docker. Note that depending on what part of Nephio you are working on, you may wish to install less or more of the components below. It should work on other environments with appropriate tweaking.
```
Comment (efiacor) [inline developer-guide/minimal-environment.md:5]: Do we need a dev machine spec table?

File: developer-guide/minimal-environment.md [hunk 14]
```diff
@@ -0,0 +1,137 @@
...
+    "configurations": [
+        {
+            "name": "Launch Nephio Controller Manager",
+            "type": "go",
+            "request": "launch",
+            "mode": "auto",
+            "program": "${workspaceFolder}/operators/nephio-controller-manager",
+            "args": [
+                "--reconcilers",
+                "*"
```
Comment (efiacor) [inline developer-guide/minimal-environment.md:125]: Might be good to point out that a subset of the reconcilers can be run here. Or "*" to run them all.

**Review by efiacor — COMMENTED** [review]
A few minor comments.

File: developer-guide/minimal-environment.md (hunk 13 above)
Comment (liamfallon) [reply developer-guide/minimal-environment.md:5]: Maybe later, I don't think we need it right now.

File: developer-guide/minimal-environment.md (hunk 14 above)
Comment (liamfallon) [reply developer-guide/minimal-environment.md:125]: done

File: developer-guide/README.md (hunk 12 above)
Comment (liamfallon) [reply developer-guide/README.md]: done

**Comment by liamfallon** [comment]
Can I get a LGTM on this please?

**Pull request merged**